lib/galaxy/webapps/galaxy/api/tags.py

```diff
@@ -0,0 +1,43 @@
+"""
+API Controller providing Galaxy Tags
+"""
+import logging
+
+from galaxy.exceptions import MessageException
+from galaxy.web import expose_api
+from galaxy.webapps.base.controller import BaseAPIController, UsesTagsMixin
+
+log = logging.getLogger(__name__)
+
+
+class TagsController(BaseAPIController, UsesTagsMixin):
+
+    # Retag an item. All previous tags are deleted and new tags are applied.
+    @expose_api
+    def update(self, trans, payload, **kwd):
+        """
+        *PUT /api/tags/
+        Apply a new set of tags to an item; previous tags are deleted.
+        """
+        item_id = payload.get("item_id")
+        item_class = payload.get("item_class")
+        item_tags = payload.get("item_tags")
+        if item_id is None:
+            raise MessageException("Please provide the item id (item_id).")
+        if item_class is None:
+            raise MessageException("Please provide the item class (item_class).")
+        if item_tags and len(item_tags) > 0:
+            item_tags = ",".join(item_tags).encode('utf-8')
```

Comment:
You'll get bytestrings here which seems wrong. Was there any reason for the encode call ?
```suggestion
            item_tags = ",".join(item_tags)
```

Reply:
Got it from: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/galaxy/controllers/tag.py#L77

Reply:
We don't want to deal with bytestrings as little as possible, and I don't see a reason to do that here. Can you update both places please ?